src/test/java/org/prebid/server/bidder/dmx/DmxBidderTest.java

```diff
@@ -88,6 +91,23 @@ public void makeHttpRequestsShouldReturnErrorWhenRequestContainsNoIdentifierIdIs
                 .containsExactly(BidderError.badInput("This request contained no identifier"));
     }
 
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenAppIdIsNullAndDeviceIfaIsAbsent() {
```

Comment:
Actually test name is not true, because if we check validateRequest, we will see that no exception will be throw if user has id